app/controllers/application_controller.rb

```diff
@@ -292,9 +292,15 @@ def access_denied(options ={})
   end
 
   def admin_only_access_denied
-    flash[:error] = ts("Sorry, only an authorized admin can access the page you were trying to reach.")
-    redirect_to root_path
-    false
+    respond_to do |format|
+      format.html do
+        flash[:error] = ts("Sorry, only an authorized admin can access the page you were trying to reach.")
+        redirect_to root_path
+        false
+      end
+      errors = [ts("Sorry, only an authorized admin can do that.")]
```

Comment:
Can you move this line inside the `format.json` block, since it's only used there?